### ISSUE TITLE
`--reuse-db` does nothing and test database is flushed

### ISSUE THREAD

**Issue by cursedgrape** [pytest-dev/pytest-django#1052]
As per the [docs]( https://pytest-django.readthedocs.io/en/latest/database.html#reuse-db-reuse-the-testing-database-between-test-runs):

> Using `--reuse-db` will create the test database in the same way as `manage.py` test usually does. However, after the test run, the test database will not be removed. The next time a test run is started with --reuse-db, the database will instantly be re used. This will allow much faster startup time for tests.

This actually doesn't work

Here's `pytest.ini`

    [pytest]
    DJANGO_SETTINGS_MODULE = myproject.settings
    addopts = --reuse-db

and here's `django_db_setup`

    import sqlite3
    import uuid
    
    import pytest
    from django.conf import settings
    from django.contrib.auth.models import User
    from django.core.management import call_command
    
    
    @pytest.fixture(scope='session')
    def test_user():
        return {
            'username': uuid.uuid4().hex[:10],
            'password': uuid.uuid4().hex,
        }
    
    
    @pytest.fixture(scope='session')
    def django_db_setup(django_db_blocker, test_user, django_db_keepdb):
        db_path = settings.BASE_DIR / 'test-db.sqlite3'
        sqlite3.connect(db_path.as_posix())
        settings.DATABASES['default']['NAME'] = db_path
        with django_db_blocker.unblock():
            call_command('migrate', '--noinput')
            User.objects.create_user(
                username=test_user['username'],
                password=test_user['password'],
                email=f'{test_user["username"]}@test',
                is_active=True,
            )

and `django_db_keepdb` evaluates to `True`. Here's the tests:

    def test1(django_user_model, django_db_keepdb):
        print(django_db_keepdb, django_user_model.objects.all())
    
    
    def test2(django_user_model, django_db_keepdb):
        print(django_db_keepdb, django_user_model.objects.all())

which results in:

    True <QuerySet [<User: b18cd8369b>]>
    True <QuerySet []>

and the expected results are

    True <QuerySet [<User: b18cd8369b>]>
    True <QuerySet [<User: b18cd8369b>]>

So, how to actually obtain this outcome?

**Comment by robertddewey** [comment]
Would like to know as well. The `--reuse-db` flag is useful to create the skeleton DB schema, but there are times where I want to actually inspect the data generated from all of the tests.

**Comment by martin-thoma** [comment]
@cursedgrape Did you find a work-around?

**Comment by cursedgrape** [comment]
> @cursedgrape Did you find a work-around?

yeah, I stopped using `pytest-django` and currently using `pytest` + handling the data however I want.

**Comment by kingbuzzman** [comment]
There is a misunderstanding here. The fixture [`django_test_environment`](https://github.com/pytest-dev/pytest-django/blob/v4.11.1/pytest_django/plugin.py#L509) is what actually creates the db.

You're overriding fixture [`django_db_setup`](https://github.com/pytest-dev/pytest-django/blob/v4.11.1/pytest_django/fixtures.py#L177) that relies on -- meaning that by the time you call `django_db_setup` is too late. Not to mention, that since you're not calling the "super()" of it, you miss out on all the django prep that should be done.

This was my test:

```python
@pytest.fixture(autouse=True, scope="session")
def django_test_environment(request):
    db_path = settings.BASE_DIR / 'test-db.sqlite3'

    settings.DATABASES['default']['ENGINE'] = 'django.db.backends.sqlite3'
    settings.DATABASES['default']['NAME'] = str(db_path)

    yield request.getfixturevalue("django_test_environment")



@pytest.fixture(scope='session')
def django_db_setup(django_db_blocker, test_user, request):
    request.getfixturevalue("django_db_setup")
    with django_db_blocker.unblock():
        call_command('migrate', '--noinput')
        User.objects.create_user(
            username=test_user['username'],
            password=test_user['password'],
            email=f'{test_user["username"]}@test',
            is_active=True,
        )
        yield
```

Works as expected:

```shell
$ runtests nest/payout/tests/test_thing.py --create-db
...

True <QuerySet [<User: 219427c3f6>]>
.True <QuerySet [<User: 219427c3f6>]>
```

```shell
$ runtests nest/payout/tests/test_thing.py --create-db
...

True <QuerySet [<User: 948060a23b>]>
.True <QuerySet [<User: 948060a23b>]>
```

---

_Note: that **948060a23b** was created in the step above_
```shell
$ runtests nest/payout/tests/test_thing.py --reuse-db
...                                                                                                                                                        

nest/payout/tests/test_thing.py True <QuerySet [<User: 948060a23b>, <User: e4bef5b5de>]>
.True <QuerySet [<User: 948060a23b>, <User: e4bef5b5de>]>
.
```

_Note: that **e4bef5b5de** was created in the step above_
```shell
$ runtests nest/payout/tests/test_thing.py --reuse-db
...                                                                                                                                                                 

nest/payout/tests/test_thing.py True <QuerySet [<User: 948060a23b>, <User: e4bef5b5de>, <User: 73d750ec6f>]>
.True <QuerySet [<User: 948060a23b>, <User: e4bef5b5de>, <User: 73d750ec6f>]>
.

---

I vote to close this issue.
